frontend/src/pages/notebookController/screens/server/StartServerModal.tsx

```diff
@@ -24,18 +41,41 @@ const StartServerModal: React.FC<StartServerModalProps> = ({
   setStartModalShown,
   onClose,
 }) => {
+  const { lastNotebookCreationTime } = React.useContext(NotebookControllerContext);
+  const { notebookNamespace } = useNamespaces();
+  const [spawnInProgress, setSpawnInProgress] = React.useState<boolean>(false);
+  const [logsExpaned, setLogsExpaned] = React.useState<boolean>(false);
+  const [spawnPercentile, setSpawnPercentile] = React.useState<number>(0);
   const [spawnStatus, setSpawnStatus] = React.useState<SpawnStatus | null>(null);
 
   const isNotebookRunning = checkNotebookRunning(notebook);
   const notebookLink = notebook?.metadata.annotations?.['opendatahub.io/link'];
 
+  const notebookEvents = useWatchNotebookEvents(
+    notebookNamespace,
+    notebook?.metadata.name || '',
+    spawnInProgress,
+  );
+  const spawnFailed =
+    spawnStatus?.status === AlertVariant.danger || spawnStatus?.status === AlertVariant.warning;
+
+  const notebookStatus = getNotebookStatus(notebookEvents, lastNotebookCreationTime);
+
+  const onUnload = React.useCallback((e: BeforeUnloadEvent) => {
+    e.preventDefault();
+    return (e.returnValue = '');
+  }, []);
+
   React.useEffect(() => {
     let timer;
     if (isNotebookRunning) {
+      window.removeEventListener('beforeunload', onUnload);
```

Comment:
Reviewing it with Juntao, this is because the redirect happens within' the hook so the other hook cannot unload fast enough to avoid the natural navigation causing the prompt. This is likely poor structuring in the hooks that is causing this weirdness... we should look at this in the future but it's needed for now.